### PR TITLE
Recycler view attempt

### PR DESCRIPTION
**Pull request by Julesssss**

### CODE REVIEW

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+
+        final ViewHolder holder;
+        final TheVergeXmlParser.Entry e = articles.get(position);
+
+        // Init view if null
+        if (null == convertView) {
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java:53]: Why null == convertView and not convertView == null? It's standard convention to put the variable you're testing first in the condition statement.

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        // Update view's widgets
+        holder.title.setText(e.title);
+        holder.content.setText(formatContentFromHtml(e.content));
+        holder.cardLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Crouton.makeText((Activity) mContext,
+                        "onClick: " + e.title.substring(0, 25), Style.INFO).show();
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java:80]: Save the 25 as a constant at the top...the variable name would give someone else an idea as to why you decided to put in that magic number!

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleListFragment.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    /**
+     * The fragment argument representing the section number for this
+     * fragment.
+     */
+    private static final String ARG_SECTION_NUMBER = "section_number";
+
+    /**
+     * Boolean to detect whether the fragment is still visable
+     */
+    public static boolean isVisable;
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleListFragment.java:35]: Typo, visible*

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleListFragment.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        mAdapter = new RecyclerAdapter(getActivity());
+        mRecyclerView.setAdapter(mAdapter);
+
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        isVisable = true;
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleListFragment.java:75]: Avoid using variables to track lifecycle states, its redundant and error prone. Android tracks most of the states...try getView(), might be the thing you need here.

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java [hunk 5]
```diff
@@ -213,6 +212,7 @@ private void loadXmlFromNetwork(String urlString) throws XmlPullParserException,
         String title = null;
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: Initializing to null is redundant

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java (hunk 1 above)
Comment (Julesssss) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleArrayAdapter.java:53]: Not sure, thought I'd saw it like that before. The file can be deleted anyway

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/RecyclerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        // each data item is just a string in this case
+        public LinearLayout mRelLayout;
+        TextView title;
+        TextView content;
+        TextView updated;
+
+        public ViewHolder(LinearLayout v, TextView t, TextView c, TextView u) {
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/RecyclerAdapter.java:38]: Don't be lazy about variable names ;) This makes code harder to read

File: app/src/main/res/layout/article_card_list_item.xml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="3dp"
+                android:background="@color/colorPrimaryLight"/>
+
+            <TextView
+                android:id="@+id/textViewContent"
+                android:layout_below="@id/divider1"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textSize="17sp"
```
Comment (amitav13) [inline app/src/main/res/layout/article_card_list_item.xml:50]: Let's start factoring out text sizes into dimens and styles, easier to keep track of that way.

**Pull request merged**